swap/src/monero.rs

```diff
@@ -1,58 +1,60 @@
 use anyhow::Result;
 use async_trait::async_trait;
 use backoff::{backoff::Constant as ConstantBackoff, future::FutureOperation as _};
+use futures::TryFutureExt;
 use monero::{Address, Network, PrivateKey};
 use monero_harness::rpc::wallet;
-use std::{str::FromStr, time::Duration};
-
+use std::time::Duration;
 use url::Url;
+
 pub use xmr_btc::monero::{
     Amount, CreateWalletForOutput, InsufficientFunds, PrivateViewKey, PublicKey, PublicViewKey,
-    Transfer, TransferProof, TxHash, WatchForTransfer, *,
+    Transfer, *,
 };
 
-pub struct Wallet(pub wallet::Client);
+pub struct Facade {
```

Comment:
IMO, `Facade` is a very generic term. Why is it not a wallet anymore?

Maybe just call it `Wallets` as I can see it wraps both the user wallet and the readonly wallet.

Reply:
Yeah, it's a bad name.